salesforce-connector/src/com/axonivy/connector/salesforce/auth/OAuth2Feature.java

```diff
@@ -0,0 +1,158 @@
+import ch.ivyteam.ivy.rest.client.oauth2.OAuth2TokenRequester.AuthContext;
+import ch.ivyteam.ivy.rest.client.oauth2.uri.OAuth2CallbackUriBuilder;
+import ch.ivyteam.ivy.rest.client.oauth2.uri.OAuth2UriProperty;
+
+public class OAuth2Feature implements Feature {
+	public static interface Default {
+		String AUTH_URI = "https://login.salesforce.com/services/oauth2/token";
+		String APP_SCOPE = "";
+		String USER_SCOPE = "read api";
+	}
+
+	public static interface Property {
+		String APP_ID = "AUTH.appId";
+		String CLIENT_SECRET = "AUTH.secretKey";
+		String SCOPE = "AUTH.scope";
+		String AUTH_BASE_URI = "AUTH.baseUri";
+		String USE_APP_PERMISSIONS = "AUTH.useAppPermissions";
+		String USE_USER_PASS_FLOW = "AUTH.userPassFlow";
+
+		String USER = "AUTH.user";
+		String PASS = "AUTH.password";
+	}
+
+	@Override
+	public boolean configure(FeatureContext context) {
+		var config = new FeatureConfig(context.getConfiguration(), OAuth2Feature.class);
+		var graphUri = new OAuth2UriProperty(config, Property.AUTH_BASE_URI, Default.AUTH_URI);
+		var oauth2 = new OAuth2BearerFilter(ctxt -> requestToken(ctxt, graphUri), graphUri);
+		//oauth2.tokenScopeProperty(Property.SCOPE);
+		oauth2.tokenSuffix(() -> GrantType.of(config).type);
+		context.register(oauth2, Priorities.AUTHORIZATION);
+		return true;
+	}
+
+	private static Response requestToken(AuthContext ctxt, OAuth2UriProperty uriFactory) {
+		FeatureConfig config = ctxt.config;
+		var authCode = ctxt.authCode();
+		var refreshToken = ctxt.refreshToken();
+		GrantType grant = GrantType.of(config);
+		if (authCode.isEmpty() && refreshToken.isEmpty() && GrantType.AUTH_CODE == grant) {
+			authError(config, uriFactory).withMessage("missing permission from user to act in his name.").throwError();
+		}
+
+		Form form = createTokenPayload(config, authCode, refreshToken);
+		var response = ctxt.target.request().accept(MediaType.WILDCARD).post(Entity.form(form));
+		return response;
+	}
+
+	static Form createTokenPayload(FeatureConfig config, Optional<String> authCode, Optional<String> refreshToken) {
+		Form form = new Form();
+		form.param("client_id", config.readMandatory(Property.APP_ID));
+		form.param("client_secret", config.readMandatory(Property.CLIENT_SECRET));
+		GrantType grant = GrantType.of(config);
+		form.param("grant_type", grant.type);
+		configureGrant(config, authCode, form, grant);
+		if (refreshToken.isPresent()) {
+			form.param("redirect_uri", ivyCallbackUri().toASCIIString());
+			form.param("refresh_token", refreshToken.get());
+			form.asMap().putSingle("grant_type", "refresh_token");
+		}
+		return form;
+	}
+
+	private static void configureGrant(FeatureConfig config, Optional<String> authCode, Form form, GrantType grant) {
+		switch (grant) {
+		case APPLICATION:
+			//form.param("scope", config.read(Property.SCOPE).orElse(Default.APP_SCOPE));
+			break;
+		case PASSWORD:
+			//form.param("scope", getPersonalScope(config));
```

Comment:
do we need this line code for investigating any issue? or can we remove it? @HoDacVu90